libraries/chain/controller.cpp

```diff
@@ -3753,17 +3803,17 @@ struct controller_impl {
          return;
       }
 
-      // Don't save the QC from block extension if the claimed block has a better valid_qc.
-      if (claimed->valid_qc && (claimed->valid_qc->is_strong() || received_qc.is_weak())) {
+      // Don't save the QC from block extension if the claimed block has a better or same valid_qc.
+      if (received_qc.is_weak() || claimed->valid_qc_is_strong()) {
```

Comment:
Maybe `claimed->valid_qc_is_strong()` should be placed the first, as strong is common.

Reply:
I placed `received_qc.is_weak()` because first because it can be checked without acquiring a mutex.

Reply:
Maybe put a comment there.